api/views/yangSearch/grep_search.py

```diff
@@ -101,43 +100,38 @@ def _get_matching_module_names(
             .replace("'", r"\'")  # noqa: Q000
             .replace(r"\\'", r"\'")  # noqa: Q000
         )
-        cache_key = hashlib.sha256(
-            f'{search_string}{inverted_search}{case_sensitive}'
-            f'{str(sorted(organizations)) if organizations else ""}'.encode(),
-        ).hexdigest()
+        cache_key = (
+            f'{search_string}{inverted_search}{case_sensitive}{str(sorted(organizations)) if organizations else ""}'
+        )
         if result := self._get_cached_search_results(cache_key):
             return result
-        pcregrep_shell_command = (
-            f'pcregrep -{"" if case_sensitive else "i"}lrMe \'{search_string}\' '
-            f'{self.all_modules_directory} | uniq '
-            '| awk -F/ \'{ print $NF }\''
+        search_command = self._get_filesystem_search_command(
+            search_string,
+            inverted_search,
+            case_sensitive,
+            organizations,
         )
-        if not inverted_search and organizations:
-            # in the case of an inverted search, organizations will be resolved in the ES search only
-            pcregrep_shell_command += f' | grep -E \'{"|".join(organizations)}\''
         try:
-            module_names_with_file_extension = (
-                subprocess.check_output(
-                    pcregrep_shell_command,
-                    shell=True,
-                )
-                .decode()
-                .split('\n')
-            )
-            if inverted_search:
-                module_names_with_file_extension = tuple(
-                    set(self._get_all_modules_with_filename_extension()) - set(module_names_with_file_extension),
-                )
-            self._cache_search_results(cache_key, module_names_with_file_extension)
-            return self._get_modules_from_cursor(module_names_with_file_extension)
-        except subprocess.CalledProcessError as e:
-            if not e.output and inverted_search:
-                self.logger.info('All the modules satisfy the inverted search')
-                return self._get_modules_from_cursor(self._get_all_modules_with_filename_extension())
-            elif not e.output:
-                self.logger.info(f'Did not find any modules satisfying such a search: {search_string}')
-                return
+            command_output, error = search_command.communicate()
+        except Exception as e:
+            self.logger.exception(f'Such a search: {search_string}, caused an error: {str(e)}')
```

Comment:
Doesn't logging and then raising again mean there will be two different errors logged for this? We should avoid that if possible. The exception type should also be more specific. I think `OSError` is the only one that could occur here.

Reply:
The only problem I see here is that the error text would be duplicated if ```exc_info``` is set to ```True```: 
![image](https://user-images.githubusercontent.com/60445869/205880200-f132d86f-8786-4811-be5b-29d0d9f287c2.png)
so I set it to ```False```, did you mean this?

Reply:
I mean that in yang.log we'll see the message explicitly logged here with `logger.exception()`, and flask will log it's own message when it catches the `ValueError`. We could just pass the `f'Such a search: {search_string}, caused an error: {error}'` string into the `ValueError` and let flask catch it. Using `logger.exception()` mostly makes sense when we don't want to propagate the exception.

Reply:
Got it, thanks, fixed